util/chunk/chunk.go

```diff
@@ -277,6 +278,70 @@ func (c *Chunk) AppendPartialRow(colIdx int, row Row) {
 	}
 }
 
+// PreAlloc pre-allocates the memory space in a Chunk to store the Row.
+// NOTE:
+// 1. The Chunk must be empty or holds no useful data.
+// 2. The schema of the Row must be the same with the Chunk.
+// 3. This API is paired with the `Insert()` function, which inserts all the
+//    rows data into the Chunk after the pre-allocation.
```

Comment:
how about adding another NOTE in the comment:
```go
// 4. We set the null bitmap here instead of in the Insert() function because
//    the Insert() function is called parallelly, the data race on a byte can
//    not be avoided although the manipulated bits are different inside a byte.
```